CPT331.iOS/CPT331.iOS/LocationViewController.swift

```diff
@@ -0,0 +1,123 @@
+    
+    @IBOutlet weak var backgroundView: UIView!
+    @IBOutlet weak var visualEffectView: UIVisualEffectView!
+    @IBOutlet weak var headerView: UIView!
+    @IBOutlet weak var containerView: UIView!
+    @IBOutlet weak var footerView: UIView!
+    
+    @IBOutlet weak var headingLabel: UILabel!
+    @IBOutlet weak var subheadingLabel: UILabel!
+    
+    @IBOutlet weak var eventsTabButton: UIButton!
+    @IBOutlet weak var crimeTabButton: UIButton!
+    @IBOutlet weak var infoTabButton: UIButton!
+    
+    @IBAction func closeButtonTapped(sender: AnyObject) {
+        self.dismissViewControllerAnimated(true, completion: {})
+    }
+    
+    @IBAction func eventsTabButtonTapped(sender: AnyObject) {
+        self.toggleSubview("locationEventsView")
+    }
+    
+    @IBAction func crimeTabButtonTapped(sender: AnyObject) {
+        self.toggleSubview("locationCrimeView")
+    }
+    
+    @IBAction func infoTabButtonTapped(sender: AnyObject) {
+        self.toggleSubview("locationInfoView")
+    }
+    
+    // Passed in from MapViewController
+    var location:Location!
+    
+    // Stores which subview is currently being displayed
+    // Used to manage view controller hierarchy
+    weak var currentSubview: UIViewController?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Round corners
+        self.visualEffectView.layer.cornerRadius = 4
+        self.visualEffectView.clipsToBounds = true
+        
+        self.headingLabel.text = location.name
+        
+        // Show events view by default
+        self.toggleSubview("locationEventsView")
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    func toggleSubview(id:String) {
+        if let vc = self.storyboard?.instantiateViewControllerWithIdentifier(id) as? LocationSubViewController {
+            
+            // Remove current child view controller (if exists)
+            if let current = self.currentSubview {
+                for view in self.containerView.subviews {
+                    view.removeFromSuperview()
+                }
+                
+                current.willMoveToParentViewController(nil)
+                current.removeFromParentViewController()
+                self.currentSubview = nil
+            }
+            
+            // Pass location info
+            vc.location = self.location
+            
+            // Add new child view controller
+            vc.view.frame = self.containerView.frame
+            self.addChildViewController(vc)
+            self.containerView.addSubview(vc.view)
+            vc.didMoveToParentViewController(self)
+            self.currentSubview = vc
+            
+            // Update button colors
+            self.toggleButton(id)
+        }
+    }
+    
+    func toggleButton(id:String) {
```

Comment:
Seems brittle to switch on some arbitrary text. Is there a way you can use an object itself? Or at least concrete up these values into a constant that is shared by the button and the function?